add v22.1.0-beta.3 release notes (#13604)
cockroachdb/docs: _includes/releases/v22.1/v22.1.0-beta.3.md

## v22.1.0-beta.3

Release Date: April 18, 2022

{% include releases/release-downloads-docker-image.md release=include.release %}

<h3 id="v22-1-0-beta-3-enterprise-edition-changes">Enterprise edition changes</h3>

> - Unified the syntax for defining the behavior of initial scans on [changefeeds](../v22.1/create-changefeed.html) by extending the `initial_scan` option to accept three possible values: `'yes|no|only'`. [#79471][#79471]

Reviewer: Yes that change is good. I have an anchor link on the initial scan option, so could change "changefeeds" to change-data-capture-overview.html and "`initial_scan`" to create-changefeeds.html#initial-scan.

> - Changefeed messages for tables with multiple [column families](../v22.1/column-families.html) using the `split_column_families` option or the [`CREATE CHANGEFEED for TABLE bar FAMILY f_1`](../v22.1/create-changefeed.html) syntax will append the family name to the table name in the default topic. For example, `CREATE CHANGEFEED FOR TABLE foo, TABLE bar FAMILY f_1` will emit to the topics `foo` and `bar.f_1`.  In cloud storage sinks such as s3, the filename will include the family separated by a `+` (e.g., `bar+f_1`). A changefeed created with the `split_column_families` option will create new topics if new families are added to a watched table that already had multiple column families. [#79448][#79448]

Reviewer: Yes, this looks good, but I actually think this could be shortened since the docs for the feature were merged last week? 

Changefeeds can now target tables with [more than one column family](use-changefeeds.html#changefeeds-on-tables-with-column-families) using either the [`split_column_families` option](create-changefeeds.html#split-column-families) or the `FAMILY` keyword . Changefeeds will emit individual messages per column family on a table.

Either way it looks good to me, here are some relevant links:
**The column family work**: use-changefeeds.html#changefeeds-on-tables-with-column-families, provides a description of the working & gotchas 
**example**: create-a-changefeed-on-a-table-with-column-families. 
**option anchor link**: create-changefeeds.html#split-column-families

- The `full_table_name` option is now supported for all [changefeed](../v22.1/change-data-capture-overview.html) sinks. [#79448][#79448]
- `LATEST` files are no longer overwritten and are now versioned and written in the `/metadata/latest` directory for non-mixed clusters. [#79553][#79553]
- Previously, the [`ALTER CHANGEFEED`](../v22.1/alter-changefeed.html) statement would not work with changefeeds that use fully qualified names in their [`CREATE CHANGEFEED`](../v22.1/create-changefeed.html) statement. This is now fixed by ensuring that each existing target is added with its fully qualified name so that it can be resolved in validation checks. Every changefeed will now display the fully qualified name of every target in the [`SHOW CHANGEFEED JOB`](../v22.1/show-jobs.html) query. [#79745][#79745]
- Added a `changefeed.backfill.scan_request_size` setting to control scan request size during backfill. [#79710][#79710]

<h3 id="v22-1-0-beta-3-sql-language-changes">SQL language changes</h3>

> - CockroachDB now ensures the user passes the same number of locality-aware URIs for the full [backup](../v22.1/take-full-and-incremental-backups.html) destination as the `incremental_location` parameter (i.e., `BACKUP INTO LATEST IN ($1, $2, $3) WITH incremental_location = ($4, $5, $6)`). [#79600][#79600]

Reviewer 2: Not a fan of i.e. and e.g. in docs. Here, it looks like it could be replaced with "for example".

> - `EXPLAIN (DDL)`, when invoked on statements supported by the declarative schema changer, prints a plan of what the schema changer will do. This can be useful for anticipating the complexity of a schema change (e.g., anything involving backfill or validation operations might be slow to run) and for troubleshooting. `EXPLAIN (DDL, VERBOSE)` produces a more detailed plan. [#79780][#79780]

Reviewer 2: ```suggestion
- `EXPLAIN (DDL)`, when invoked on statements supported by the declarative schema changer, prints a plan of what the schema changer will do. This can be useful for anticipating the complexity of a schema change (for example, anything involving backfill or validation operations might be slow to run) and for troubleshooting. `EXPLAIN (DDL, VERBOSE)` produces a more detailed plan. [#79780][#79780]
```

<h3 id="v22-1-0-beta-3-operational-changes">Operational changes</h3>

- Added a new time-series metric, `storage.marked-for-compaction-files`, for the count of files marked for compaction. This is useful for monitoring storage-level background migrations. [#79370][#79370]

> - [Changefeed](../v22.1/change-data-capture-overview.html) creation and failure event logs are now emitted to the [`TELEMETRY` logging channel](../v22.1/eventlog.html#telemetry-events). [#79749][#79749]

Reviewer 2: ```suggestion
- [Changefeed](../v22.1/change-data-capture-overview.html) creation and failure event logs are now emitted to the [`TELEMETRY`](../v22.1/eventlog.html#telemetry-events) logging channel. [#79749][#79749]
```

<h3 id="v22-1-0-beta-3-command-line-changes">Command-line changes</h3>

> - Introduced a new `ttllogger` [workload](../v22.1/cockroach-workload.html) which creates a TTL table emulating a "log" with rows expiring after the duration specified in the `--ttl` flag. [#	][#79482]

Reviewer 2: ```suggestion
- Introduced a new `ttllogger` [workload](../v22.1/cockroach-workload.html) which creates a TTL table emulating a "log" with rows expiring after the duration specified in the `--ttl` flag. [#79482][#79482]
```

<h3 id="v22-1-0-beta-3-db-console-changes">DB Console changes</h3>

> - Filtering by column has been added to the [Hot Ranges page](../v22.1/ui-hot-ranges-page.html). [#79647][#79647]

Reviewer 2: ```suggestion
- The [Hot Ranges page](../v22.1/ui-hot-ranges-page.html) now allows filtering by column. [#79647][#79647]
```

- Fixed a bug that caused an internal error when the inner expression of a column access expression evaluated to `NULL`. For example, evaluation of the expression `(CASE WHEN b THEN ((ROW(1) AS a)) ELSE NULL END).a` would error when `b` is `false`. This bug was present since v19.1 or earlier. [#79529][#79529]
- Fixed a bug that caused an error when accessing a named column of a labeled tuple. The bug only occurred when an expression could produce one of several different tuples. For example, `(CASE WHEN true THEN (ROW(1) AS a) ELSE (ROW(2) AS a) END).a` would fail to evaluate. This bug was present since v22.1.0. Although present in previous versions, it was impossible to encounter due to limitations that prevented using tuples in this way. [#79529][#79529]
- Previously, queries reading from an index or primary key on `FLOAT` or `REAL` columns `DESC` would read `-0` for every `+0` value stored in the index. This has been fixed to correctly read `+0` for `+0` and `-0` for `-0`. [#79533][#79533]

> - Separate tenants now use unique values for the internal field `ClusterID`. Previously, separate tenants that were hosted on the same storage cluster would be assigned the same `ClusterID`. [#79742][#79742]

Reviewer 2: ```suggestion
```

Drop because we are no longer including multi-tenant architecture features in release notes.

- Fixed some cases where a job or schema change that had encountered an error would continue to execute for some time before eventually failing. [#79713][#79713]
- Previously, the optional `is_called` parameter of the `setval` function would default to `false` when not specified. It now defaults to `true` to match PostgreSQL behavior. [#79779][#79779]

> - Updated props on the [Raft Messages](../v22.1/ui-debug-pages.html) page to include functions to set the time window and time scale in order to fix the date picker and drag-to-zoom functionality. [#79791][#79791]

Reviewer 2: ```suggestion
- Updated props on the [Raft Messages](../v22.1/ui-debug-pages.html) page to include functions to set the time window and time scale in order to fix the date picker and drag-to-zoom functionality. [#79791][#79791]
```

I think the mention of props is internal, regarding how the page is built, and this alternative would be sufficient.

```suggestion
- On the [Raft Messages](../v22.1/ui-debug-pages.html) page, the date picker and drag-to-zoom functionality are now fixed. [#79791][#79791]
```

> - Fixed a bug where Pebble compaction heuristics could allow a large compaction backlog to accumulate, eventually triggering high read amplfiication. [#79597][#79597]

Reviewer 2: ```suggestion
- Fixed a bug where Pebble compaction heuristics could allow a large compaction backlog to accumulate, eventually leading to high read amplification. [#79597][#79597]
```
